File: ERP/backend/django-ninja/src/apps/cadastro/services/produto_service.py
# ./src/apps/cadastro/services/produto_service.py

from ..data.produto_data import ProdutoData

class ProdutoService:
	@staticmethod
	def inserir(data):
		return ProdutoData.inserir(data)

	@staticmethod
	def buscar_todos():
		return ProdutoData.buscar_todos()

	@staticmethod
	def buscar_por_id(id):
		return ProdutoData.buscar_por_id(id)

	@staticmethod
	def alterar(id, data):
		produto = ProdutoData.buscar_por_id(id)
		return ProdutoData.alterar(produto, data)

	@staticmethod
	def excluir(id):
		produto = ProdutoData.buscar_por_id(id)
		ProdutoData.excluir(produto)

File: ERP/frontend/frontend_angular/src/app/services/produto.service.ts
// ./src/app/services/produto.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Produto } from '../models/produto';

@Injectable({
  providedIn: 'root'
})
export class ProdutoService {
  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) { }

  buscar(texto?: string): Observable<Produto[]> {
    if (texto) {
      return this.http.get<Produto[]>(`${this.apiUrl}?search=${texto}`);
    }
    return this.http.get<Produto[]>(this.apiUrl);
  }

  buscarPorId(id: number): Observable<Produto> {
    return this.http.get<Produto>(`${this.apiUrl}${id}`);
  }

  inserir(produto: Produto): Observable<Produto> {
    return this.http.post<Produto>(this.apiUrl, produto);
  }

  alterar(id: number, produto: Produto): Observable<Produto> {
    return this.http.put<Produto>(`${this.apiUrl}${id}`, produto);
  }

  excluir(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}${id}`);
  }
}


